packages/dashboard/e2e/tests/widgets/text.spec.ts

```diff
@@ -280,5 +280,38 @@ test.describe('Test Text Widget', () => {
       await page.getByText(TEXT_WIDGET_CONTENT).click();
       await page.waitForURL(`**${url}`);
     });
+
+    test('when link is enabled, it should sanitize href and not link if dangerous', async ({
+      dashboardWithTextWidget,
+      configPanel,
+      page,
+    }) => {
+      const widget = dashboardWithTextWidget.gridArea.locator(
+        '[data-gesture=widget]'
+      );
```

Comment:
[Out of scope]: I do not think this is a good selector. `.locator` should be used as a last resort and I do not understand `[data-gesture=widget]`. If I was a user, I would not find my text widget by looking for `[data-gesture=widget]`. Using a selector like this is brittle and disconnected from how a user uses the dashboard. We should improve this. I realize the other tests are using this, so no need to address here.